persist/sqlite/addresses.go

```diff
@@ -10,6 +10,35 @@ import (
 	"go.sia.tech/walletd/v2/wallet"
 )
 
+// CheckAddresses returns true if any of the addresses have been seen on the
+// blockchain. This is a quick way to scan wallets for lookaheads.
+//
+// If the index mode is not full, this function will only return true if
+// an address is registered with a wallet.
+func (s *Store) CheckAddresses(addresses []types.Address) (known bool, err error) {
+	err = s.transaction(func(tx *txn) error {
+		stmt, err := tx.Prepare(`SELECT true FROM sia_addresses WHERE sia_address=$1`)
```

Comment:
any reason you didn't go for `SELECT 1 FROM sia_addresses WHERE sia_address = ANY($1) LIMIT 1`?

Reply:
Probably not supported by the driver

Reply:
`ANY()` is special to Postgres.